flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/ParquetPojoInputFormat.java

```diff
@@ -0,0 +1,133 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A subclass of {@link ParquetInputFormat} to read from Parquets files and convert to POJO type.
+ * It is mainly used to read simple data type without nested fields.
+ */
+public class ParquetPojoInputFormat<E> extends ParquetInputFormat<E> {
+
+	private static final Logger LOG = LoggerFactory.getLogger(ParquetPojoInputFormat.class);
+	private final Class<E> pojoTypeClass;
+	private final TypeSerializer<E> typeSerializer;
+	private transient Field[] pojoFields;
+
+	public ParquetPojoInputFormat(Path filePath, MessageType messageType, PojoTypeInfo<E> pojoTypeInfo) {
+		super(filePath, messageType);
+		this.pojoTypeClass = pojoTypeInfo.getTypeClass();
+		this.typeSerializer = pojoTypeInfo.createSerializer(new ExecutionConfig());
+		final Map<String, Field> fieldMap = new HashMap<>();
+		findAllFields(pojoTypeClass, fieldMap);
+		selectFields(fieldMap.keySet().toArray(new String[fieldMap.size()]));
+	}
+
+	@Override
+	public void open(FileInputSplit split) throws IOException {
+		super.open(split);
+		pojoFields = new Field[getFieldNames().length];
+		LOG.error("Fields number is %d", getFieldNames().length);
+		System.out.println(String.format("Fields number is %d", getFieldNames().length));
+		final Map<String, Field> fieldMap = new HashMap<>();
+		findAllFields(pojoTypeClass, fieldMap);
+
+		for (int i = 0; i < getFieldNames().length; ++i) {
+			String fieldName = getFieldNames()[i];
+			pojoFields[i] = fieldMap.get(fieldName);
+
+			if (pojoFields[i] != null) {
+				pojoFields[i].setAccessible(true);
+			} else {
+				throw new RuntimeException(
+					String.format("There is no field called %s in %s", fieldName, pojoTypeClass.getName()));
+			}
+		}
+	}
+
+	private void findAllFields(Class<?> clazz, Map<String, Field> fieldMap) {
+
+		for (Field field : clazz.getDeclaredFields()) {
+			fieldMap.put(field.getName(), field);
+		}
+
+		if (clazz.getSuperclass() != null) {
+			findAllFields(clazz.getSuperclass(), fieldMap);
+		}
+	}
+
+	@Override
+	protected E convert(Row row) {
+		E result = typeSerializer.createInstance();
+		for (int i = 0; i < row.getArity(); ++i) {
+			try {
+				if (pojoFields[i].getType().isAssignableFrom(List.class)) {
+					pojoFields[i].set(result, Arrays.asList(row.getField(i)));
+				} else {
+					pojoFields[i].set(result, row.getField(i));
+				}
+			} catch (IllegalAccessException e) {
+				throw new RuntimeException(
+					String.format("Parsed value could not be set in POJO field %s", getFieldNames()[i]));
+			}
+		}
+
+		return result;
+	}
+
+	/**
+	 * Extracts the {@link TypeInformation}s  from {@link PojoTypeInfo} according to the given field name.
+	 */
+	private static <E> TypeInformation<?>[] extractTypeInfos(PojoTypeInfo<E> pojoTypeInfo, String[] fieldNames) {
```

Comment:
This private method extractTypeInfos(...) is not used anywhere within this ParquetPojoInputFormat class. Is it redundant and should be removed?